link_linux.go

```diff
@@ -851,6 +851,14 @@ func (h *Handle) linkModify(link Link, flags int) error {
 		msg.Change |= syscall.IFF_MULTICAST
 		msg.Flags |= syscall.IFF_MULTICAST
 	}
+	if base.Index != 0 {
+		if flags&syscall.NLM_F_CREATE == 0 {
```

Comment:
I don't think it should check for NLM_F_CREATE and error here. Specifically this will break BridgeSetMcastSnoop if you attempt to retrieve the link (which sets the index value) and then use the retrieved link to set snoop.

Reply:
Let me remove that check. I was just being paranoid. I did not realize that there was a case beyond create where it was valid to set the index.